### PR TITLE
Improve release process documentation

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR changes the internal release documentation to be more intuitive and be up-to-date with some other tools we use.

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/14001

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: doc/internal/release/how-to-release.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ That includes:
 - **Making sure the people doing the release have access to all the tools and infrastructure needed to do the release.**
   > - This includes write access to the Vitess repository and to the Maven repository.
 - **Preparing and cleaning the release notes summary.**
+  > - If the release does not contain significant change (i.e. a small patch release) then this step can be skipped
```
Comment (deepthi) [inline doc/internal/release/how-to-release.md:61]:
```suggestion
  > - If the release does not contain significant changes (i.e. a small patch release) then this step can be skipped
```

File: doc/internal/release/how-to-release.md [hunk 2]
```diff
@@ -77,19 +78,15 @@ That includes:
...
-  > - We open this Pull Request now to avoid waiting on the CI during release day.
-  > - All future changes to the release notes during the code freeze will need to be ported to both PRs: the one on `main` and the Release Pull Request.
 - **Update the website documentation.**
   > - We want to open a preparatory **draft** Pull Request to update the documentation.
   > - There are several pages we want to update:
-      >   - [The releases page](https://vitess.io/docs/releases/), we must add the new release to the list with all its information and link. The links can be broken (404 error) while we are preparing for the release, this is fine.
-  >   - [The local install page](https://vitess.io/docs/get-started/local/), we must use the proper version increment for this guide and the proper SHA. The SHA will have to be modified once the Release Pull Request and the release is tagged is merged.
+  >  - [The releases page](https://vitess.io/docs/releases/), we must add the new release to the list with all its information and link. The links can be broken (404 error) while we are preparing for the release, this is fine.
```
Comment (deepthi) [inline doc/internal/release/how-to-release.md:84]:
```suggestion
  >  - [The releases page](https://vitess.io/docs/releases/): we must add the new release to the list with all its information and link. The links can be broken (404 error) while we are preparing for the release, this is fine.
```

File: doc/internal/release/how-to-release.md [hunk 3]
```diff
@@ -77,19 +78,15 @@ That includes:
...
+  >  - [The local install page](https://vitess.io/docs/get-started/local/), we must use the proper version increment for this guide and the proper SHA. The SHA will have to be modified once the Release Pull Request and the release is tagged is merged.
```
Comment (deepthi) [inline doc/internal/release/how-to-release.md:85]:
```suggestion
  >  - [The local install page](https://vitess.io/docs/get-started/local/): we must use the proper version increment for this guide and the proper SHA. The SHA will have to be modified once the Release Pull Request and the release is tagged is merged.
```

File: doc/internal/release/how-to-release.md [hunk 4]
```diff
@@ -77,19 +78,15 @@ That includes:
...
   > - If we are doing a GA or RC release follow the instructions below:
-      >  - There are two scripts in the website repository in `./tools/{ga|rc}_release.sh`, use them to update the website documentation. The scripts automate:
-           >    - For an RC, we need to create a new version in the sidebar and mark the current version as RC.
-  >    - For a GA, we need to mark the version we are releasing as "Stable" and the next one as "Development".
+  >  - There are two scripts in the website repository in `./tools/{ga|rc}_release.sh`, use them to update the website documentation. The scripts automate:
+  >  - For an RC, we need to create a new version in the sidebar and mark the current version as RC.
```
Comment (deepthi) [inline doc/internal/release/how-to-release.md:88]:
```suggestion
  >  - For an RC, we need to create a new version in the sidebar and mark it as RC.
```

File: doc/internal/release/how-to-release.md [hunk 5]
```diff
@@ -102,7 +99,7 @@ On the release day, there are several things to do:
 - **Tag the Vitess release.**
   > - A guide on how to tag a version is available in the [How To Release Vitess](#how-to-release-vitess) section.
 - **Update the release notes on `main`.**
-  > - During the code freeze, we created a Pull Request against `main` to update the release notes. It must be merged.
+  > - One Pull Request against `main` must be created, it will contain the new release notes that we are adding in the Release Pull Request.
```
Comment (deepthi) [inline doc/internal/release/how-to-release.md:102]: This is confusing. Are there two pull requests? One that this step is asking you to create, and another one that you are referring to as "Release Pull Request"?

File: doc/internal/release/how-to-release.md [hunk 6]
```diff
@@ -125,10 +122,10 @@ On the release day, there are several things to do:
   > - The benchmarks need to complete before announcing the blog posts or before they get cross-posted.
 - **Go back to dev mode on the release branch.**
   > - The version constants across the codebase must be updated to `SNAPSHOT`.
-- **Build k8s Docker images and publish them**
+- **Build k8s Docker images and publish them.**
   > - The docker image for `base`, `lite`, etc are built automatically by DockerHub. The k8s images however are dependent on these images and are required to be built manually.
   > - These images should be built after the `base` image has been built and available on DockerHub.
-  > - To build and publish these images, run `./release.sh` from the directory `vitess/docker`.
+  > - To build and publish these images, checkout to the new `git tag` and run `./release.sh` from the directory `vitess/docker`.
```
Comment (deepthi) [inline doc/internal/release/how-to-release.md:128]:
```suggestion
  > - To build and publish these images, checkout the new `git tag` and run `./release.sh` from the directory `vitess/docker`.
```

Comment (deepthi) [reply doc/internal/release/how-to-release.md:128]: `git tag` looks like a command. Maybe you mean something like:

- To build and publish these images, checkout the new release tag that was just created and run `./release.sh` from the directory `vitess/docker`.

File: doc/internal/release/how-to-release.md [hunk 7]
```diff
@@ -172,15 +169,12 @@ We need to verify that _arewefastyet_ has finished the benchmark too.
...
-    2. Run the following command to generate the release notes:
-        1. Release Candidate:
-            ```shell
-            go run ./go/tools/release-notes --from "v14.0.3" --to "HEAD" --version "v15.0.0-rc1" --summary "./changelog/15.0/15.0.0/summary.md" [--threads=[0-9.]]
-            ```
-        2. General Availability:
-            ```shell
-            go run ./go/tools/release-notes --from "v14.0.3" --to "HEAD" --version "v15.0.0" --summary "./changelog/15.0/15.0.0/summary.md" [--threads=[0-9.]]
-            ```
+    2. Run the following command to generate the release notes, please note that you can omit the `--summary` flag if there are no summary.
```
Comment (deepthi) [inline doc/internal/release/how-to-release.md:172]:
```suggestion
    2. Run the following command to generate the release notes. Note that you can omit the `--summary` flag if there are no summary.
```

**Review by deepthi — APPROVED** [review]
I made several suggestions. It will be good to incorporate them, but I'm giving you an approval anyway.

File: doc/internal/release/how-to-release.md (hunk 4 above)
Comment (frouioui) [reply doc/internal/release/how-to-release.md:88]: Changing `the current version` to `it` would imply that we are changing the new version in the sidebar to `RC`, which is incorrect. Let's assume we are releasing `v18.0.0-rc1`. We want to create a new version in the sidebar (`v19.0`), and mark the current version (the one we are releasing, `v18.0`) as `RC`. I will change the wording to make it more clear :)

File: doc/internal/release/how-to-release.md (hunk 5 above)
Comment (frouioui) [reply doc/internal/release/how-to-release.md:102]: Yes there are. There is a "Release Pull Request" which is merged on the release branch. And there is another one which is made against `main` to update the release notes.

File: doc/internal/release/how-to-release.md (hunk 6 above)
Comment (frouioui) [reply doc/internal/release/how-to-release.md:128]: Yes that's what I mean, i'll change it

File: doc/internal/release/how-to-release.md (hunk 4 above)
Comment (frouioui) [reply doc/internal/release/how-to-release.md]: This is fixed via 136c780568

File: doc/internal/release/how-to-release.md (hunk 6 above)
Comment (frouioui) [reply doc/internal/release/how-to-release.md]: This is fixed via 4d509fc185

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**